thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/DataAvailabilityTaskScheduler.java

```diff
@@ -89,12 +101,16 @@ public void run() {
         long detectionConfigId = detectionConfig.getId();
         if (!runningDetection.containsKey(detectionConfigId)) {
           if (isAllDatasetUpdated(detectionConfig, detection2DatasetMap.get(detectionConfig), dataset2RefreshTimeMap)) {
-            //TODO: additional check is required if detection is based on aggregated value across multiple data points
-            createDetectionTask(detectionConfig);
-            ThirdeyeMetricsUtil.eventScheduledTaskCounter.inc();
-            taskCount++;
+            if (isWithinSchedulingWindow(detection2DatasetMap.get(detectionConfig), dataset2RefreshTimeMap)) {
+              //TODO: additional check is required if detection is based on aggregated value across multiple data points
+              createDetectionTask(detectionConfig);
+              ThirdeyeMetricsUtil.eventScheduledTaskCounter.inc();
+              taskCount++;
+            } else {
+              LOG.warn("Unable to schedule a task for {}, because it is out of scheduling window.", detectionConfigId);
```

Comment:
Should we log it? Will it creates lots of logs?

Reply:
It will only generate logs when the scheduler runs, namely every 5 minutes. This line will only be printed when the watermark is not moving forward, which should be minor case.

Reply:
That's fine. Thanks.